piptools/utils.py

```diff
@@ -249,3 +249,15 @@ def temp_environ():
     finally:
         os.environ.clear()
         os.environ.update(environ)
+
+
+def make_line_from_ireq(ireq):
+    """Return a single requirement line from an InstallRequirement"""
+    line = format_requirement(ireq)
+
+    ireq_hashes = ireq.options.get('hashes', dict())
+    for algorithm, hexdigests in ireq_hashes.items():
+        for hash_ in hexdigests:
+            line += " \\\n    --hash={}:{}".format(algorithm, hash_)
```

Comment:
This block of code looks extremely similar to `_format_requirement` in `writer.py`. Any chance this could be factored out to some common utility function?

Reply:
Agreed. I could update `piptools.utils.format_requirement` to render hashes.

Reply:
Please look a the last commit. I've fixed it.